agents-manage-ui/src/components/form/json-schema-input.tsx

```diff
@@ -33,16 +43,49 @@ export function JsonSchemaInput<T extends FieldValues>({
       description={description}
       isRequired={isRequired}
     >
-      {(field) => (
-        <StandaloneJsonEditor
-          placeholder={placeholder}
-          {...field}
-          value={field.value || ''} // can be `null`
-          onChange={field.onChange}
-          readOnly={readOnly}
-          disabled={disabled}
-        />
-      )}
+      {(field) => {
+        const value = field.value || ''; // can be `null`
+        return (
+          <StandaloneJsonEditor
+            placeholder={placeholder}
+            {...field}
+            value={value}
+            onChange={field.onChange}
+            readOnly={readOnly}
+            disabled={disabled}
+            actions={
+              <Dialog defaultOpen>
+                <DialogTrigger asChild>
+                  <Button
+                    type="button"
+                    variant="outline"
+                    size="sm"
+                    className="backdrop-blur-xl h-6 px-2 text-xs rounded-sm"
+                  >
+                    Simple Edit
+                  </Button>
+                </DialogTrigger>
+                <DialogContent className="!max-w-4xl">
+                  <DialogHeader>
+                    <DialogTitle>Structured output (JSON)</DialogTitle>
+                    <DialogDescription>
+                      The model will generate a JSON object that matches this schema.
+                    </DialogDescription>
+                    <JsonSchemaBuilder value={value} onChange={field.onChange} />
+                  </DialogHeader>
+                  {/*<DialogFooter>*/}
+                  {/* TODO */}
+                  {/*<Button variant="secondary">*/}
+                  {/*  <SquarePenIcon />*/}
+                  {/*  Generate*/}
+                  {/*</Button>*/}
+                  {/*</DialogFooter>*/}
```

Comment:
added for https://linear.app/inkeep/issue/PRD-5114/json-schema-can-be-created-or-edited-in-natural-language